libs/ml/examples/bert_simple_cls_example/main.cpp

```diff
@@ -0,0 +1,307 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/tensor.hpp"
+#include "ml/core/graph.hpp"
+#include "ml/layers/fully_connected.hpp"
+#include "ml/layers/normalisation/layer_norm.hpp"
+#include "ml/layers/self_attention_encoder.hpp"
+#include "ml/ops/add.hpp"
+#include "ml/ops/embeddings.hpp"
+#include "ml/ops/loss_functions/cross_entropy_loss.hpp"
+#include "ml/ops/slice.hpp"
+#include "ml/optimisation/adam_optimiser.hpp"
+
+#include <iostream>
+#include <string>
+
+using namespace fetch::ml::ops;
+using namespace fetch::ml::layers;
+
+using DataType   = float;
+using TensorType = fetch::math::Tensor<DataType>;
+using SizeType   = typename TensorType::SizeType;
+using SizeVector = typename TensorType::SizeVector;
+
+using GraphType     = typename fetch::ml::Graph<TensorType>;
+using StateDictType = typename fetch::ml::StateDict<TensorType>;
+using OptimiserType = typename fetch::ml::optimisers::AdamOptimiser<TensorType>;
+
+using RegType         = fetch::ml::RegularisationType;
+using WeightsInitType = fetch::ml::ops::WeightsInitialisation;
+using ActivationType  = fetch::ml::details::ActivationType;
+
+struct BERTConfig
+{
+  // the default config is for bert base uncased pretrained model
```

Comment:
can you give a link to where to download the dataset you're using too - you've set a specific vocab size after all

Reply:
that would be included in the fetch-script. cuz the conversion from pytorch pretrained model to txt C++ readable file happens there